### PR TITLE
Ethnic evaluation

### PR DESCRIPTION
**Pull request by Sengxian**

### CODE REVIEW

File: evaluation/dataset.py [hunk 1]
```diff
@@ -15,6 +15,10 @@
 
 from .configs import BaseConfig, MultiChoiceTaskConfig, GenerationTaskConfig, LanguageModelTaskConfig
 from .utils import get_tokenized_input
+import torch
+def print_rank_0(*args, **kwargs):
```
Comment (Sengxian) [inline evaluation/dataset.py:19]: from .utils import print_rank_0

File: .gitignore [hunk 2]
```diff
@@ -3,3 +3,6 @@ __pycache__
 samples
 .DS_Store
 .idea
+logs/
```
Comment (Sengxian) [inline .gitignore:6]: No need to update this file

File: evaluation/__init__.py [hunk 3]
```diff
@@ -1,6 +1,6 @@
 from .configs import *
 from .model import ModelForEvaluation
-from .tasks import BaseTask, GenerationTask, MultiChoiceTask, LanguageModelTask
+from .tasks import BaseTask, GenerationTask, MultiChoiceTask, LanguageModelTask, StereoSetTask, CrowsPairTask
```
Comment (Sengxian) [inline evaluation/__init__.py:3]: Move task-specific Task to `tasks` folder.

File: evaluation/dataset.py [hunk 4]
```diff
@@ -224,22 +231,37 @@ def build_multiple_choice_sample(text, choices, is_single_token, unified_multita
...
             choice_target_id.append(np.arange(len(token), len(token) + len(choice), dtype=np.int64))
             attention_mask.append(np.tril(np.ones((len(choice), len(choice)), dtype=np.int64)))
             token = np.concatenate((token, [sop_id], choice[:-1]))
             target = np.concatenate((target, choice))
 
             if is_single_token:
                 break
-
+        
+        #if len(token)!=len(position_id) or len(position_id)!=len(attention_mask):
```
Comment (Sengxian) [inline evaluation/dataset.py:263]: Delete those comments

File: evaluation/tasks.py [hunk 5]
```diff
@@ -204,8 +204,191 @@ def build_dataset(self, relative_path):
...
+                result_dict_group = (result_list, len(dataset))
+
+            result_dict_all[group_name] = result_dict_group
+            
+        print_rank_0(f"Evaluation results of task {self.config.name}:")
+        if self.verbose:
+            for group_name, result_dict_group in result_dict_all.items():
+                self.report_group_metrics(group_name, result_dict_group)
+
+        print_rank_0(f"Finish task {self.config.name} in {time.time() - start:.1f}s.")
```
Comment (Sengxian) [inline evaluation/tasks.py:391]: Can we reuse code rather than make a copy?

File: configs/model_glm_130b.sh [hunk 6]
```diff
@@ -1,5 +1,5 @@
 MODEL_TYPE="glm-130b"
-CHECKPOINT_PATH="<your checkpoint path>"
+CHECKPOINT_PATH="<your evaluation dataset base directory>"
```
Comment (Sengxian) [inline configs/model_glm_130b.sh:2]: Do not modify this file

File: evaluation/metrics.py [hunk 7]
```diff
@@ -88,4 +123,23 @@ def calculate_perplexity(loss: List[float], data):
...
+        "EM": qa_exact_match,
+        "F1": qa_f1,
+        "Accuracy": accuracy_metric,
+        "PPL": calculate_perplexity,
+        "Precision": precision_metric,
+        "Recall": recall_metric,
+    }
+)
+ADD_METRICS = {"F1_mul": F1_metric}
+# ADD_METRICS = {"F1_mul":F1_metric,"SS_ICAT":StereoSet_ICAT,"CP":CrowsPair}
```
Comment (Sengxian) [inline evaluation/metrics.py:143]: Remove this line

File: evaluation/model.py [hunk 8]
```diff
@@ -113,7 +114,7 @@ def cond_log_prob(self, batch) -> List[List[float]]:
                 log_probs_single = []
                 for choice, choice_target_id in zip(choices, choice_target_ids):
                     tmp = output[choice_target_id, choice]
-                    log_probs_single.append(tmp.sum().tolist())
+                    log_probs_single.append(tmp.mean().tolist())
```
Comment (Sengxian) [inline evaluation/model.py:117]: Move this normalization to task's `predict_single_batch`.

File: tasks/ethnic/tasks.py [hunk 9]
```diff
@@ -0,0 +1,390 @@
+import os
```
Comment (Sengxian) [inline tasks/ethnic/tasks.py:1]: Remove this duplicate file.
